raft/server.cc

```diff
@@ -692,7 +701,7 @@ future<> server_impl::do_on_leader_with_retries(seastar::abort_source* as, Async
 
     while (true) {
         if (as && as->abort_requested()) {
-            throw request_aborted();
+            throw request_aborted(format("Request aborted while doing retries on current leader: {}, previous leader: {}", leader, prev_leader));
```

Comment:
"while doing retries" is too generic and not accurate:
- first of all, this could happen in the first attempt, not during retry.
- `do_on_leader_with_retries` is just a utility to perform `action` and retry if it fails.

We should print what action we were trying.